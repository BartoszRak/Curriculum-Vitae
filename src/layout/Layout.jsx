import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'

import Footer from './Footer'
import Sidebar from './Sidebar'
import Greeting from './Greeting'

import Home from '~pages/Home'

import style from './Layout.style'

export const Layout = ({ classes }) => (
  <React.Fragment>
    <div className={classes.root}>
      <Greeting />
      <Switch>
        <Route component={Home} to="/" />
        <Redirect from="*" to="/" />
      </Switch>
      <Footer />
    </div>
  </React.Fragment>
)

Layout.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Layout)
