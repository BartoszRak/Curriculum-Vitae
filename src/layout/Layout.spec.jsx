import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { Layout } from './Layout'

jest.unmock('@material-ui/core')

describe('component: Layout', () => {
  afterEach(cleanup)

  describe('rendering', () => {
    test('render without crash', () => {
      const wrapper = shallow(<Layout />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot', () => {
      const wrapper = shallow(<Layout />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
