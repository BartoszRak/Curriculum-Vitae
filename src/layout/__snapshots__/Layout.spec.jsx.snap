// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Layout should render as expected 1`] = `
<Fragment>
  <InjectIntl(Navigation) />
  <div>
    <GreetingMock />
    <SwitchMock>
      <RouteMock
        component="HomeMock"
        to="/"
      />
      <RedirectMock
        from="*"
        to="/"
      />
    </SwitchMock>
    <FooterMock />
  </div>
  <div
    aria-hidden={true}
  />
  <aside>
    <FormattedMessage
      id="layout.information.title"
      values={Object {}}
    />
  </aside>
</Fragment>
`;
